storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -138,7 +129,12 @@ func (s *fileSystemSource) enqueueReaders(
 
 		readers := make([]fs.FileSetReader, 0, len(files))
 		for i := 0; i < len(files); i++ {
-			r := readerPool.get()
+			r, err := readerPool.get()
+			if err != nil {
+				s.log.Errorf("unable to get reader from pool")
+				readersCh <- shardReaders{err: err}
```

Comment:
do you need a `readerPool.put(r)` after this?

Reply:
Ah no because reader is nil here (the err was from trying to get it from the pool hehe).